src/olympia/scanners/tasks.py

```diff
@@ -267,18 +267,19 @@ def run_yara_query_rule(query_rule_pk):
         return
     log.info('Fetching versions for run_yara_query_rule on rule %s', rule.pk)
     # Build a huge list of all pks we're going to run the tasks on.
-    pks = Version.unfiltered.all(
+    qs = Version.unfiltered.all(
     ).filter(
         addon__type=amo.ADDON_EXTENSION,
         files__is_webextension=True,
-    ).exclude(
-        addon__status=amo.STATUS_DISABLED,
-    ).values_list('id', flat=True).order_by('pk')
+    )
+    if not rule.run_on_disabled_addons:
+        qs = qs.exclude(addon__status=amo.STATUS_DISABLED)
+    qs = qs.values_list('id', flat=True).order_by('pk')
```

Comment:
Nit: This was called `pks` previously, I assume because it's a list of pks and not a queryset. It is still a list of pks here, so maybe it should still be called `pks` here and on line 282?

Reply:
I never liked to call this `pks` : it is, at this point, technically still a queryset. It just so happens that it's a *values* queryset, containing only pks, but because we can still apply filters etc on it at this point (and we do if not excluding disabled add-ons), I preferred to rename it to `qs`. If I had assigned the `list(qs)` to a variable I would have named that variable `pks`.